src/components/SignUpForm/index.tsx

```diff
@@ -0,0 +1,318 @@
+  // Email Change Event
+  const onChangeEmail = (e: SyntheticEvent<HTMLInputElement>) => {
+    setEmail(e.currentTarget.value);
+    setIsValidEmail(false);
+  };
+
+  // Mbti Change Event
+  const onChangeMbti = (e: SyntheticEvent<HTMLSelectElement>) => {
+    setMbti(e.currentTarget.value);
+    if (e.currentTarget.value !== '1') {
+      setIsValidMbti(false);
+    }
+  };
+
+  // Gender Change Event
+  const onChangeGender = (e: SyntheticEvent<HTMLSelectElement>) => {
+    setGender(e.currentTarget.value);
+  };
+
+  // LoginId onBlur Event
+  const onBlurLoginId = () => {
+    if (!loginId) {
+      setIsValidLoginId(true);
+    }
+  };
+
+  // Nickname onBlur Event
+  const onBlurNickname = () => {
+    if (!nickname) {
+      setIsValidNickname(true);
+    }
+  };
+
+  // Password onBlure Event
+  const onBlurPassword = () => {
+    if (!password) {
+      setIsValidPassword(true);
+      setPasswordErrorMessage(PW_REQUIRE_CHECK);
+      return;
+    }
+
+    if (password.length <= 8) {
+      setIsValidPassword(true);
+      setPasswordErrorMessage(PW_VALID_CHECK);
+    }
+  };
+
+  // PasswordConfirm onBlur Event
+  const onBlurPasswordConfirm = () => {
+    if (!passwordConfirm) {
+      setIsValidPasswordConfirm(true);
+      setPasswordConfirmErrorMessage(PWCONFIRM_REQUIRE_CHECK);
+      return;
+    }
+
+    // 비밀번호가 일치하지 않다면
+    if (password !== passwordConfirm) {
+      setIsValidPasswordConfirm(true);
+      setPasswordConfirmErrorMessage(PWCONFIRM_VALID_CHECK);
+    }
+  };
+
+  // Email onBlur Event
+  const onBlurEmail = () => {
+    if (!email) {
+      setIsValidEmail(true);
+      setEmailErrorMessage(EMAIL_REQUIRE_CHECK);
+      return;
+    }
+
+    // 이메일 유효성 검사
+    if (!regEmail.test(email)) {
+      setIsValidEmail(true);
+      setEmailErrorMessage(EMAIL_VALID_CHECK);
+    }
+  };
+
+  // MBTI onBlur Event
+  const onBlurMbti = () => {
+    if (mbti === '1') {
+      setIsValidMbti(true);
+    }
+  };
+
+  // form sign up Check button
+  const handleLastCheckClick = (e: React.MouseEvent<HTMLButtonElement>): void => {
+    e.preventDefault();
+    if (isValidLoginId === false && !loginId) {
+      setIsValidLoginId(true);
+    } else if (isValidNickname === false && !nickname) {
+      setIsValidNickname(true);
+    } else if (isValidPassword === false && !password) {
+      setIsValidPassword(true);
+    } else if (
+      isValidPasswordConfirm === false &&
+      !passwordConfirm &&
+      password !== passwordConfirm
+    ) {
+      setIsValidPasswordConfirm(true);
+    } else if (isValidEmail === false && !email) {
+      setIsValidEmail(true);
+    } else if (isValidMbti === false && mbti === '1') {
+      setIsValidMbti(true);
+    } else {
+      // 회원가입 버튼 눌렀을때
+      // 콘솔 value
+      console.log(
+        `로그인아이디 : ${loginId}
+        닉네임:${nickname}
+        패스워드:${password}
+        패스워드확인:${passwordConfirm}
+        이메일:${email}
+        mbti:${mbti} 
+        gender:${gender}`
+      );
+    }
+  };
```

Comment:
```suggestion
const handleLastCheckClick = (e: React.MouseEvent<HTMLButtonElement>): void => {
  e.preventDefault();
  switch (false) {
    case isValidLoginId:
      if (!loginId) {
        setIsValidLoginId(true);
      }
      break;
    case isValidNickname:
      if (!nickname) {
        setIsValidNickname(true);
      }
      break;
    case isValidPassword:
      if (!password) {
        setIsValidPassword(true);
      }
      break;
    case isValidPasswordConfirm:
      if (!passwordConfirm || password !== passwordConfirm) {
        setIsValidPasswordConfirm(true);
      }
      break;
    case isValidEmail:
      if (!email) {
        setIsValidEmail(true);
      }
      break;
    case isValidMbti:
      if (mbti === '1') {
        setIsValidMbti(true);
      }
      break;
    default:
// 그 뒤로 작성
  }
};
```

리팩토링 하기 애매한거 같아요.
그나마 switch 문으로 작성해봤는데 가독성면에선 더 좋을 듯 싶습니다.